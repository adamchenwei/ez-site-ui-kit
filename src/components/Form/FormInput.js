import React, { Component } from 'react'; import PropTypes from 'prop-types';
import ComponentWrapper from './../ComponentWrapper/ComponentWrapper';
const setComponentWrapperContainerClasses = require('./../../util/setup/setComponentWrapperContainerClasses');
const setComponentWrapperStyle = require('./../../util/setup/setComponentWrapperStyle');

export default class FormInput extends Component {

  constructor(props) {
    super(props);
  }

  render() {

    let {
      content,
      style,
      type,
      name,
    } = this.props;
    const COMPONENT_NAME = 'FormInput';
    const inlineStyle = setComponentWrapperStyle(style);
    const containerLevelStyle = setComponentWrapperContainerClasses(COMPONENT_NAME);
    return (
      <ComponentWrapper
        containerLevelClass={containerLevelStyle}
        style={inlineStyle}>
        <input type="text" name="name" />
      </ComponentWrapper>
    );
  }
}

FormInput.propTypes = {
  content: PropTypes.object,
  style: PropTypes.object,
  type: PropTypes.string,
  name: PropTypes.string,
};