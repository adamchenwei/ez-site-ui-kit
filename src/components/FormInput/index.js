import React, { Component } from 'react';
import PropTypes from 'prop-types';
import Container from './style/Container';
import FormLegend from './style/FormLegend';
import InputField from './style/InputField';
import FormTextarea from './style/FormTextarea';

export default class FormInput extends Component {
  constructor(props) {
    super(props);
    const {
      inputType,
    } = props;
    this.getInputField = this.getInputField.bind(this);
    this.getTextArea = this.getTextArea.bind(this);
    let InputComponent = null;
    switch (inputType) {
      case 'text-area':
        InputComponent = this.getTextArea;
        break;
      case 'email':
      case 'text':
      case 'phone':
        InputComponent = this.getInputField;
        break;
      default:
        break;
    }
    this.InputComponent = InputComponent;
  }

  getInputField() {
    const {
      fieldLegendTitle,
      fieldName,
      inputType,
      placeholderText,
    } = this.props;
    return (
      <InputField
        name={fieldName}
        gridAreaId={inputType}
        type={inputType}
        placeholder={placeholderText}
      />
    );
  }
  getTextArea() {
    const {
      fieldLegendTitle,
      fieldName,
      inputType,
      placeholderText,
    } = this.props;

    return (
      <FormTextarea
        gridAreaId={inputType}
        name={fieldName}
        placeholder={placeholderText}
      />
    );
  }
  render() {
    const {
      fieldLegendTitle,
      fieldName,
      inputType,
      placeholderText,

    } = this.props;
    const InputComponent = this.InputComponent;
    return (
      <Container>
        <FormLegend gridAreaId={inputType}>{fieldLegendTitle}</FormLegend>
        {InputComponent ? <InputComponent /> : null}
      </Container>
    );
  }
}

FormInput.propTypes = {
  fieldLegendTitle: PropTypes.string,
  fieldName: PropTypes.string,
  inputType: PropTypes.string,
  placeholderText: PropTypes.string,
};

FormInput.defaultProps = {
  fieldLegendTitle: '',
  fieldName: '',
  inputType: '',
  placeholderText: '',
};
