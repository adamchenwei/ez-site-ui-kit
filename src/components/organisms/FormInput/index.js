import React from 'react';
import PropTypes from 'prop-types';
// import Input, { InputLabel } from 'material-ui/Input';
// import { FormControl } from 'material-ui/Form';
import { Input, Label, FormGroup } from 'reactstrap';
import omit from 'lodash/omit';

function FormInput(props) {
  const {
    id,
    title,
  } = props;

  const inputProps = omit(props, [
    'id',
    'title',
  ]);
  return (
    <FormGroup>
      <Label
        htmlFor={id}
      >
        {title}
      </Label>
      <Input
        {
          ...inputProps
        }
      />
    </FormGroup>
  );
}

FormInput.propTypes = {
  id: PropTypes.string,
  title: PropTypes.string,
  onChange: PropTypes.func.isRequired,
  value: PropTypes.string,
  name: PropTypes.string,
  type: PropTypes.string,
  placeholder: PropTypes.string,
};

FormInput.defaultProps = {
  id: '',
  title: '',
  name: 'aTestInputField',
  value: '',
  type: 'text',
  placeholder: 'your text here',
};

export default FormInput;
