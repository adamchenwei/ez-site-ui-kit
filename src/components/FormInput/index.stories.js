import FormInput from './index';
import { storiesOf } from '@storybook/react';
import { action } from '@storybook/addon-actions';
import React from 'react';

storiesOf('FormInput', module)
  .add('default', () => <FormInput />)
  .add('email', () =>
    (<FormInput
      fieldLegendTitle="Your Email"
      fieldName="emailInput"
      inputType="email"
      placeholderText="i.e. mohammad@gmail.com"
    />))
  .add('text', () =>
    (<FormInput
      fieldLegendTitle="Your Text"
      fieldName="textInput"
      inputType="text"
      placeholderText="text here"
    />))
  .add('text-area', () =>
    (<FormInput
      fieldLegendTitle="Your Text"
      fieldName="textInput"
      inputType="text-area"
      placeholderText="text here"
    />));
