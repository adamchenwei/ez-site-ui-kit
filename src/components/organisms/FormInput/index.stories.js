import React from 'react';
import PropTypes from 'prop-types';
import { storiesOf } from '@storybook/react';
import FormInput from './index';
import OrganismName from '../../../storybookonly/CategoryName/OrganismName';
import ezlogger from '../../../util/debug/ezlogger';
import onChange from './service/onChange';

class FormInputInteractive extends React.Component {
  constructor(props) {
    super(props);
    this.state = {};
    this.state[props.inputProperties.name] = props.inputProperties.value;
    this.onChange = onChange.bind(this);
  }

  render() {
    const {
      inputProperties,
      id,
      title,
    } = this.props;
    ezlogger(this.state);
    return (
      <FormInput
        {...inputProperties}
        id={id}
        title={title}
        onChange={this.onChange}
        value={this.state[inputProperties.name]}
      />
    );
  }
}

FormInputInteractive.propTypes = {
  inputProperties: PropTypes.objectOf(PropTypes.any),
  id: PropTypes.string,
  title: PropTypes.string,
};

FormInputInteractive.defaultProps = {
  inputProperties: {
    name: 'testInputField',
    value: '',
    type: 'text',
    placeholder: '',
  },
  id: '',
  title: '',
};

storiesOf(`Component/${OrganismName}/FormInput/Stand Alone`, module)
  .add('interactive: text', () => (
    <FormInputInteractive
      id="testField"
      title="Form Input Title"
      inputProperties={{
        name: 'aTestInputField',
        value: '',
        type: 'text',
        placeholder: 'your text here',
      }}
    />
  ));
