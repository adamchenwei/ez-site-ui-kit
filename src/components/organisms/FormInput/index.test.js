import React from 'react';
import renderer from 'react-test-renderer';
import FormInput from './index';

describe('FormInput', () => {
  test('renders as expected', () => {
    const tree = renderer.create(<FormInput />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
