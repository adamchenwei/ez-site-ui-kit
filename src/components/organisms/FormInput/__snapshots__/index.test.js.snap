// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FormInput renders as expected 1`] = `
<div
  className="form-group"
>
  <label
    className=" form-control-label"
    htmlFor=""
  >
    
  </label>
  <input
    className="form-control"
    name="aTestInputField"
    placeholder="your text here"
    type="text"
    value=""
  />
</div>
`;
